src/main/java/io/datanerds/avropatch/serialization/CustomTypeSerializer.java

```diff
@@ -0,0 +1,148 @@
+package io.datanerds.avropatch.serialization;
+
+import io.datanerds.avropatch.Patch;
+import io.datanerds.avropatch.schema.CustomTypes;
+import io.datanerds.avropatch.schema.CustomTypes.BigDecimalType;
+import io.datanerds.avropatch.schema.CustomTypes.BigIntegerType;
+import io.datanerds.avropatch.schema.CustomTypes.DateType;
+import io.datanerds.avropatch.schema.CustomTypes.UuidType;
+import io.datanerds.avropatch.value.conversion.AvroData;
+import org.apache.avro.Schema;
+import org.apache.avro.io.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static io.datanerds.avropatch.schema.PrimitiveTypes.*;
+
+@ThreadSafe
+public final class CustomTypeSerializer {
```

Comment:
That's neat, but the IDE doesn't know that this is the only way to de-/serialize a ```Patch``` :P Hence we need to expose the class and its functionality...